drivers/ocean/ocean.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2014 Caleb Brose, Chris Fogerty, Rob Sheehy, Zach Taylor, Nick Miller
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ocean
+
+import (
+    "fmt"
+    "sync"
+    "time"
+    "net/http"
+    "io/ioutil"
+    "encoding/json"
+
+    "github.com/lighthouse/beacon/structs"
+)
+
+var oceanToken = ""
+
+var Driver = &structs.Driver {
+    Name: "ocean",
+    IsApplicable: IsApplicable,
+    GetVMs: GetProjectVMs,
+}
+
+type DropletNetwork struct {
+    IP string `json:"ip_address,omitempty"`
+    Gateway string `json:"gateway,omitempty"`
+    Type string `json:"type,omitempty"`
+}
+
+type DropletNetworks struct {
+    V4 []*DropletNetwork `json:"v4,omitempty"`
+    V6 []*DropletNetwork `json:"v6,omitempty"`
+}
+
+type Droplet struct {
+    Name string `json:"name,omitempty"`
+    Networks DropletNetworks `json:"networks,omitempty"`
+}
+
+type DropletList struct {
+    Droplets []*Droplet
+}
+
+func GetOceanToken() string {
+    if oceanToken == "" {
+        request, _ := http.NewRequest("GET",
+            "http://169.254.169.254/metadata/v1/user-data", nil)
+
+        client := http.Client{}
+        resp, err := client.Do(request)
+        defer resp.Body.Close()
+        if err != nil {
+            return ""
+        }
+
+        body, err := ioutil.ReadAll(resp.Body)
+        if err != nil {
+            return ""
+        }
+
+        oceanToken = string(body)
+    }
+
+    return oceanToken
+}
+
+func IsApplicable() bool {
+    request, _ := http.NewRequest("GET",
+        "http://169.254.169.254/metadata/v1/", nil)
```

Comment:
Yup, https://developers.digitalocean.com/metadata/.  It'a actually the same address for [ec2](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html) & [gce](https://cloud.google.com/compute/docs/metadata) metadata.